### DISCUSSION TITLE
RootUID

### DISCUSSION THREAD

**Discussion by travisdk** [fo-dicom/fo-dicom#1836] [General]
Dear all,
Why is it that the root uid (found in DicomUID) is set to:         
public static string RootUID { get; set; } = "1.2.826.0.1.3680043.2.1343.1";
As far as I can tell it points to UK company Medical Connections?
https://oid-rep.orange-labs.fr/get/1.2.826.0.1.3680043
Thanks in advance for any input. 
Henrik Juhl / Visiana ApS

**Comment by gofal** [comment]
Late reply, but maybe still helpful.
Uids have to be globally unique. DICOM wants to achive this by having this multipart number (https://dicom.nema.org/dicom/2013/output/chtml/part05/chapter_9.html). A company or organization can request a UID and then this "root-uid" is registered and no one else may use this number. And whoever has such an root-uid is responsible that it is then used in a way, that all generated UIDs are unique. For example if a company has such an RootUid, it then can add a kind of customer-id as the next part. and then on every server add a server-id, and then as last part add a timestamp or an other identifier.

The company Medical Connections https://www.medicalconnections.co.uk/ is thankfully very open minded to the developer community. They provide a free test-server https://dicomserver.co.uk/ and they offer a free UID registration, you can get a sub-ID of their own RootID for free.

There I registered one for fo-dicom development. 
If you have a software that is widely installed and creates a lot of new data and has to generate lots of new UIDs, then it might be a good idea to register a own RootUID.

And in case you do not have such rootuids but need to generate globally unique UIDs, then DICOM has a mechanism similary to GUID by using the root 2.25. and then having a 39 digit random number https://dicom.nema.org/dicom/2013/output/chtml/part05/sect_B.2.html
